anchor-service/src/merkle_tree.rs

```diff
@@ -0,0 +1,83 @@
+use crate::anchor::{AnchorRequest, MerkleNode, MerkleNodes};
+use anyhow::{anyhow, Result};
+use ceramic_core::SerializeExt;
+use cid::Cid;
+
+pub struct MerkleTree {
+    pub root_cid: Cid,
+    pub nodes: MerkleNodes,
+    pub count: u64,
+}
+
+/// Make a tree using Merkle mountain range
+/// Ref: https://eprint.iacr.org/2021/038.pdf
+pub fn build_merkle_tree(anchor_requests: &[AnchorRequest]) -> Result<MerkleTree> {
```

Comment:
is there any unit test coverage of this?  The MMR logic here is complex enough that I think some targeted unit tests would make me feel a lot better about it.

Reply:
We were in the middle of merging this PR when we saw this comment :(

Create a ticket under SA Ph1 for this:
https://linear.app/3boxlabs/issue/AES-343/add-more-exhaustive-merkle-tree-tests